todo_app/app.py

```diff
@@ -97,43 +111,64 @@ def login_callback():
         name = github_user['login']
         user = User(name)
         login_user(user)
+        app.logger.info("User %s logged in" % name)
         return redirect(url_for('index'))
 
     @app.route('/add_item', methods=['POST'])
     @check_writer_role
     @login_required
     def add_todo_item():
-        item_title = request.form.get('title')
-        item_description = request.form.get('description')
-        mongo.add_item(item_title, item_description)
+        try:
+            item_title = request.form.get('title')
+            item_description = request.form.get('description')
+            mongo.add_item(item_title, item_description)
+            app.logger.info("Added todo item %s" % request.form)
+        except Exception as err:
+            app.logger.error(err, exc_info=True)
         return redirect(url_for('index'))
 
     @app.route('/complete/<item_id>')
     @check_writer_role
     @login_required
     def complete_item(item_id):
-        mongo.update_item(item_id, CardStatus.DONE)
+        try:
+            mongo.update_item(item_id, CardStatus.DONE)
+            app.logger.info("Completed item %s" % item_id)
+        except Exception as err:
+            app.logger.error(err, exc_info=True)
         return redirect(url_for('index'))
 
     @app.route('/do/<item_id>')
     @check_writer_role
     @login_required
     def do_item(item_id):
-        mongo.update_item(item_id, CardStatus.DOING)
+        try:
+            mongo.update_item(item_id, CardStatus.DOING)
+            app.logger.info("Update item %s status to DOING" % item_id)
+        except Exception as err:
+            app.logger.error(err, exc_info=True)
         return redirect(url_for('index'))
 
     @app.route('/uncomplete/<item_id>')
     @check_writer_role
     @login_required
     def uncomplete_item(item_id):
-        mongo.update_item(item_id, CardStatus.TODO)
+        try:
+            mongo.update_item(item_id, CardStatus.TODO)
+            app.logger.info("Update item %s status to TODO" % item_id)
+        except Exception as err:
+            app.logger.error(err, exc_info=True)
         return redirect(url_for('index'))
 
     @app.route('/delete/<item_id>')
     @check_writer_role
     @login_required
     def remove_item(item_id):
-        mongo.delete_item(item_id)
+        try:
+            mongo.delete_item(item_id)
+            app.logger.info("Deleted item %s" % item_id)
+        except Exception as err:
+            app.logger.error(err, exc_info=True)
```

Comment:
Instead of duplicating this Exception handling in each route, I think Flask lets you register "error handlers".